pages/_app.js

```diff
@@ -1,36 +1,33 @@
-import Head from 'next/head'
+import Layer0RUM from 'rum'
 import '@/styles/globals.css'
-import Header from '@/components/Header'
-import { useEffect, useState } from 'react'
+import { useEffect } from 'react'
+import Navbar from '@/components/Navbar'
+import { relativizeURL } from '@/components/Shimmer'
 import { prefetch } from '@layer0/prefetch/window/prefetch'
 
-function MyApp({ Component, pageProps }) {
-  const [mounted, setMounted] = useState('print')
+// Include the RUM Analytics in the production build only
+if (process.env.NODE_ENV === 'production') {
+  Layer0RUM('8016af1c-fb31-45c1-b1e0-312bffabbe93')
+}
+
+const MyApp = ({ Component, pageProps }) => {
   useEffect(() => {
-    setMounted('all')
-    // register a listener for SW messages to prefetch images from the PLP API responses
+    // Register a listener for SW messages to prefetch images
     const { serviceWorker } = navigator
     if (serviceWorker) {
       serviceWorker.addEventListener('message', (event) => {
         if (event.data.action === 'prefetch') {
-          prefetch(event.data.url, event.data.as, event.data.options)
+          prefetch(relativizeURL(event.data.url), event.data.as, event.data.options)
```

Comment:
It looks like your service worker is already grabbing images from the API responses. What is the reason to add this?

Reply:
Nope, the service worker doesn't prefetch the images directly. One needs to add a prefetcher to handle the prefetch logic. This is missing in Layer0 Docs as well.

Reference from where I got it https://github.com/layer0-docs/layer0-nextjs-example/blob/master/sw/service-worker.js#L14,
Internal Task: https://app.clickup.com/t/4205457/APPOPS-13285

Reply:
The `deepFetchPlugin` won't do that? 😵‍💫 

Sure seems like the way the docs are written that it should. 

```
new DeepFetchPlugin([
      // query the PLP API response for images to prefetch
      // prefetch logic is handled in _app.js
      {
        jsonQuery: 'pageProps.products.picture',
        maxMatches: 10,
        as: 'image',
      },
    ]),
```